client_test.go

```diff
@@ -317,6 +318,23 @@ func testLoopOutSuccess(ctx *testContext, amt btcutil.Amount, hash lntypes.Hash,
 
 	signalPrepaymentResult(nil)
 
+	// One spend notifier is registered by batcher in order to notify swap.
+	ctx.AssertRegisterSpendNtfn(confIntent.PkScript)
+
+	// One spend notifier is registered by batch to watch primary sweep.
+	ctx.AssertRegisterSpendNtfn(confIntent.PkScript)
+
+	// We need to notify for a new block height because that's when the
+	// batch tries to publish a new transaction. We need to call the lnd
+	// mock twice, because both executor and batcher are watching the epoch
+	// ntfns.
+	err := ctx.Lnd.NotifyHeight(ctx.Lnd.Height + 1)
```

Comment:
Are we sure this is safe? Notifying twice could trigger just the executor or the batcher too right?

Reply:
Urh right. I wonder why this didn't cause flakes?

Fixing this would require a small refactor of this mocking interface (allow for multiple subscribers to `NotifyHeight`)

Reply:
Aha right, but then again, I think that it may cause flakes so it'd probably be better to notify all subscribers from the mock separately.